src/database/mod.rs

```diff
@@ -85,7 +85,11 @@ pub trait DatabaseConnection {
     ) -> Option<(bip32::ChildNumber, bool)>;
 
     /// Get all our coins, past or present, spent or not.
-    fn coins(&mut self, coin_type: CoinType) -> HashMap<bitcoin::OutPoint, Coin>;
+    fn coins(
+        &mut self,
+        statuses: Option<&[CoinStatus]>,
+        outpoints: Option<&[bitcoin::OutPoint]>,
```

Comment:
What is `Some([])` supposed to mean? If it's not necessary, then let's drop the `Option` and use the empty list as describing the absence of filters?

Reply:
Oh yes I agree, we can remove the `Option` as the empty list achieves the same thing. I'll change that.

For the `listcoins` CLI args, should we still allow the user to omit the args completely or would we require them to pass an empty list?

If we allow to omit args, I don't think it's possible currently to pass only the second optional arg without the first, e.g.
```
listcoins --outpoints "[\"abcd:1\"]"
```
We would need to do
```
listcoins "[]" "[\"abcd:1\"]"
```